internal/envoy/v3/route.go

```diff
@@ -796,12 +797,20 @@ func queryParamMatcher(queryParams []dag.QueryParamMatchCondition) []*envoy_rout
 
 // containsMatch returns a HeaderMatchSpecifier which will match the
 // supplied substring
-func containsMatch(s string) *envoy_route_v3.HeaderMatcher_StringMatch {
+func containsMatch(s string, ignoreCase bool) *envoy_route_v3.HeaderMatcher_StringMatch {
 	// convert the substring s into a regular expression that matches s.
 	// note that Envoy expects the expression to match the entire string, not just the substring
 	// formed from s. see [projectcontour/contour/#1751 & envoyproxy/envoy#8283]
-	regex := fmt.Sprintf(".*%s.*", regexp.QuoteMeta(s))
+	var regex string
```

Comment:
this could implemented as `contains` api in Envoy as well, happy to make that change

Reply:
We should create an issue to track this so that it doesn't get lost.

Reply:
done: https://github.com/projectcontour/contour/issues/5572